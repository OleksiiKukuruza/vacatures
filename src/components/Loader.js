import React from 'react';
import PropTypes from 'prop-types';
import { PacmanLoader } from 'react-spinners';
import styled, { withTheme } from 'styled-components';

const StyledLoader = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const ThemedLoader = ({ theme }) => (
  <StyledLoader>
    <PacmanLoader color={theme.primary.main} />
  </StyledLoader>
);

ThemedLoader.propTypes = {
  theme: PropTypes.shape({
    primary: PropTypes.shape({
      main: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

const Loader = withTheme(ThemedLoader);

export default Loader;
