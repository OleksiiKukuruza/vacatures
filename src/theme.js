const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  spacingUnit: 8
};

export default theme;
