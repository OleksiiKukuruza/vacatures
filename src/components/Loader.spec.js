import React from 'react';
import { shallow } from 'enzyme';
import { ThemedLoader } from './Loader';
import theme from '../theme';

const setup = () => {
  const wrapper = shallow(
    <ThemedLoader theme={theme} />
  );

  return {
    wrapper
  };
};

describe('Loader', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
