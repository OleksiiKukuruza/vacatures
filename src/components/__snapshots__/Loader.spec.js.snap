// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders without crashing 1`] = `
<styled.div>
  <onlyUpdateForKeys(Loader)
    color="#54b6c8"
    loaderStyle={Object {}}
    loading={true}
    margin="2px"
    size={25}
    sizeUnit="px"
  />
</styled.div>
`;
